### PR TITLE
add first version of the privacy policy texts in german & english

### LINKED ISSUE #352: Legal: add privacy policy text

### LINKED ISSUE #406: Legal: Update Privacy Policy to have changeable address + email + phone number
- [x] Default value: undefined
- [x] Make address, email & phone number changeable

### PR DESCRIPTION
**Pull request by Gr33ndev**
**Totally has to be verified by the legal department..**

Closes Gamify-IT/issues#352
Closes Gamify-IT/issues#406

@madebyTimo Has to be added to the landing-page above "Third Party Licenses" with a text like: "By clicking 'Go to Login' you accept our privacy policy".

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+sonar.projectKey=privacy-policy
```
Comment (delvh) [inline sonar-project.properties:1]: We want to show this project on SonarQube?
Isn't that going to just be hanging around just like the `third-party-licenses` repo?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Privacy Policy
+
+In this repo, a web page will be provided on which the privacy policy text is displayed in german & english.
```
Comment (delvh) [inline README.md]:
```suggestion
This repo stores the privacy policy text for Gamify-IT in german and english.
```

File: index.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+<h3>Data privacy</h3>
+<p>The operators of these pages take the protection of your personal data very seriously. We treat your personal data confidentially and in accordance with the statutory data protection regulations and this privacy policy.</p>
+<p>When you use this website, various personal data are collected. Personal data is data with which you can be personally identified. This privacy policy explains what data we collect and what we use it for. It also explains how and for what purpose this is done.</p>
+<p>We point out that data transmission over the Internet (eg communication by e-mail) security gaps. A complete protection of the data against access by third parties is not possible.</p>
+<h3>Note on the responsible body</h3>
+<p>The responsible party for data processing on this website is:</p>
+<p>Institute for Software Engineering, Dept. of Software Quality & Architecture at the University of Stuttgart
+    Universitätsstraße 38, 70569 Stuttgart</p>
+<p>Phone: <a href="tel:+4971168588273">+49 711 685 88273</a> <br>
+    E-mail: <a href="mailto:steffen.becker@iste.uni-stuttgart.de">steffen.becker@iste.uni-stuttgart.de</a></p>
```
Comment (delvh) [inline index.html]: 1. Bad idea to put personal emails statically into the text. That opens up the recipient to spam bot attacks.
2. The responsible party is not always Professor Becker. It is only Professor Becker when he hosts it. It's always the one hosting it.
3. This means we need a mechanism for inserting variables into the HTML file. PHP?
4. I recommend a force-push to get rid of these explicit addresses.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+</ul>
+<p>If you have restricted the processing of your personal data, this data may - apart from being stored - only be processed with your consent or for the assertion, exercise or defense of legal claims or for the protection of the rights of another natural or legal person or for reasons of an important public interest of the European Union or a Member State.</p>
+<h3>SSL or TLS encryption</h3>
+<p>This site uses SSL or TLS encryption for security reasons and to protect the transmission of confidential content, such as orders or requests that you send to us as the site operator. You can recognize an encrypted connection by the fact that the address line of the browser changes from "http://" to "https://" and by the lock symbol in your browser line.</p>
+<p>If SSL or TLS encryption is activated, the data you transmit to us cannot be read by third parties.</p>
+
+<h2>3. Data collection on this website</h2>
+<h3>Cookies</h3>
```
Comment (delvh) [inline index.html]: We use cookies?
At least not at the moment.

File: index.html [hunk 5]
```diff
@@ -0,0 +1,89 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Privacy Policy</title>
+</head>
+<body>
+<h1>Privacy Policy</h1>
+<a href="de.html">Deutsche Version hier</a>
```
Comment (delvh) [inline index.html]:
```suggestion
<a href="de.html">&#x1F1E9;&#x1F1EA;</a>
```
To display 🇩🇪.

File: de.html [hunk 6]
```diff
@@ -0,0 +1,69 @@
+<!DOCTYPE html>
+<html lang="de">
+<head>
+    <meta charset="UTF-8">
+    <title>Datenschutzerklärung</title>
+</head>
+<body>
+<h1>Datenschutzerklärung</h1>
+<a href="index.html">English version here</a>
```
Comment (delvh) [inline de.html]:
```suggestion
<a href="index.html">&#x1F1EC;&#x1F1E7;/&#x1F1FA;&#x1F1F8</a>
```
To display `🇬🇧/🇺🇸`

**Review by delvh — CHANGES_REQUESTED** [review]

File: index.html (hunk 3 above)
Comment (Gr33ndev) [reply index.html]: We are unfortunately obliged to include concrete data, so we can't get rid of them completely + the data is 1 to 1 as static as here as on the uni website which is why we don't need a force push in my opinion. 

https://www.iste.uni-stuttgart.de/institute/contact/

File: index.html (hunk 4 above)
Comment (Gr33ndev) [reply index.html]: I thought that the login data is stored in cookies...

File: sonar-project.properties (hunk 1 above)
Comment (Gr33ndev) [reply sonar-project.properties:1]: Any repo without problems improves the overall sonar statistic

File: index.html (hunk 4 above)
Comment (madebyTimo) [reply index.html]: Yes keycloak uses cookies. But we currently don't have third-party cookies

**Comment by madebyTimo** [comment]
The link on the landing page will be added in Gamify-IT/issues#384

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,6 @@
+FROM nginx:alpine as production-build
```
Comment (delvh) [inline Dockerfile]:
```suggestion
FROM nginx:alpine AS production-build
```

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+
+COPY index.php /usr/share/nginx/html
+COPY de.php /usr/share/nginx/html
```
Comment (delvh) [inline Dockerfile]: `COPY config.json` into a volume?

**Comment by madebyTimo** [comment]
- The html code inside the php files looks fine without seeing it in a browser, but why do you use php?
    - nginx does not includes php so the docker image is not working. We would need to use apache or a php additional to nginx.
- The action for the development containers is missing.

Contact me if you need help.

File: index.php [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    <li>Browser type and version</li>
+    <li>Operating system used</li>
+    <li>Referrer URL</li>
+    <li>Host name of the accessing computer</li>
+    <li>Time of the server request</li>
+    <li>IP address</li>
+</ul>
+<p>This data is not merged with other data sources.</p>
+<p>The collection of this data is based on Art. 6 para. 1 lit. f DSGVO. The website operator has a legitimate interest in the technically error-free presentation and optimization of its website - for this purpose, the server log files must be collected.</p>
+<p>Source: <a href="https://www.datenschutzerklaerung.de">https://www.datenschutzerklaerung.de</a></p>
```
Comment (delvh) [inline index.php]: Still a link to the german site?

Comment (madebyTimo) [reply index.php]: I think this is a link to the generator and is the source of the german and the english text

**Comment by madebyTimo** [comment]
@Gr33ndev Could can you please check if the changes are fine

**Review by madebyTimo — APPROVED** [review]

File: index.php (hunk 9 above)
Comment (Gr33ndev) [reply index.php]: The German text is generated by this site, the English one is translated by deepl

**Pull request merged**